src/fidesops/api/v1/endpoints/connection_type_endpoints.py

```diff
@@ -27,41 +29,55 @@
 logger = logging.getLogger(__name__)
 
 
-def get_connection_types(search: Optional[str] = None) -> List[str]:
+def get_connection_types(search: Optional[str] = None) -> List[ConnectionSystemTypeMap]:
     def is_match(elem: str) -> bool:
         """If a search query param was included, is it a substring of an available connector type?"""
         return search in elem if search else True
 
-    database_types: List[str] = [
-        conn_type.value
-        for conn_type in ConnectionType
-        if conn_type
-        not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
-        and is_match(conn_type.value)
-    ]
-    saas_types: List[str] = [
-        saas_type.value
-        for saas_type in SaaSType
-        if saas_type != SaaSType.custom and is_match(saas_type.value)
-    ]
-    connection_types: List[str] = sorted(database_types + saas_types)
+    connection_system_types: List[ConnectionSystemTypeMap] = []
+    database_types: List[str] = sorted(
+        [
+            conn_type.value
+            for conn_type in ConnectionType
+            if conn_type
+            not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
+            and is_match(conn_type.value)
+        ]
+    )
+    connection_system_types.extend(
+        [
+            ConnectionSystemTypeMap(identifier=item, type=SystemType.database)
+            for item in database_types
+        ]
+    )
+    saas_types: List[str] = sorted(
+        [
+            saas_type.value
+            for saas_type in SaaSType
+            if saas_type != SaaSType.custom and is_match(saas_type.value)
```

Comment:
This endpoint will never return custom SaaSType connections, should we also include those?

Reply:
On the initial version of this endpoint, we had a discussion with Adria that we're not returning custom connector at this time as an option, https://github.com/ethyca/fidesops/pull/795#discussion_r914117170, the UI isn't yet supporting that flow.

Reply:
I wanted to follow the pattern we had before, where we're only including pre-registered / existing connectors for now, see https://github.com/ethyca/fidesops/pull/768#discussion_r911470444